ui/nuxeo-document-layout.js

```diff
@@ -217,8 +217,15 @@ import { I18nBehavior } from './nuxeo-i18n-behavior.js';
     }
 
     _buildHrefFn(tmpl) {
-      // eslint-disable-next-line no-new-func
-      return new Function(['document', 'layout'], `return \`${tmpl}\`.toLowerCase();`);
+      return () => {
+        const matches = tmpl.matchAll(/\${([^}]+)}/g);
```

Comment:
I would still go with limiting this to the document and layout properties (and perhaps even update the `hrefTemplate` docs to mention this). Otherwise, we can make `hrefTemplate` refer private properties and even itself? `href-template= '${hrefTemplate}-layout.html"`

Reply:
yes, I just felt it was not worth the effort as there's no sec issue but I can do it. what should I do in case the prop does not start with `layout` or `document, use an empty value or not perform the replacement?

Reply:
Maybe using an empty value would be more expectable?

Reply:
done 👍